src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,76 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { actions as currentTodoActions } from '../../features/currentTodo';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
 
 export const TodoModal: React.FC = () => {
+  const { currentTodo } = useAppSelector(state => state.currentTodo);
+  const dispatch = useAppDispatch();
+  const [userInModal, setUserInModal] = useState<User | null>(null);
+  const [isModal, setIsModal] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const handleCloseModal = () => {
+    setIsModal(false);
+    dispatch(currentTodoActions.removeTodo());
+  };
+
+  useEffect(() => {
+    if (currentTodo) {
+      setIsModal(true);
+      setIsLoading(true);
+      getUser(currentTodo.userId).then(user => {
+        setUserInModal(user);
+        setIsLoading(false);
+      });
```

Comment:
```suggestion
      })
      .finally(() => {
        setIsLoading(false);
      })
```
use Finally branch for resetting loader. It should be reseted in any scenario. so In case your request is failed you won't get endless loader